db/bigtable_eth1.go

```diff
@@ -3286,22 +3317,9 @@ func (bigtable *Bigtable) GetMetadataForAddress(address []byte) (*types.Eth1Addr
 		return nil, err
 	}
 
+	// sort balances based on token address (required for proper pagination)
 	sort.Slice(ret.Balances, func(i, j int) bool {
-		priceI := decimal.New(0, 0)
-		priceJ := decimal.New(0, 0)
-		if len(ret.Balances[i].Metadata.Price) > 0 {
-			priceI = decimal.NewFromBigInt(new(big.Int).SetBytes(ret.Balances[i].Metadata.Price), 0)
-		}
-		if len(ret.Balances[j].Metadata.Price) > 0 {
-			priceJ = decimal.NewFromBigInt(new(big.Int).SetBytes(ret.Balances[j].Metadata.Price), 0)
-		}
-
-		mulI := decimal.NewFromFloat(float64(10)).Pow(decimal.NewFromBigInt(new(big.Int).SetBytes(ret.Balances[i].Metadata.Decimals), 0))
-		mulJ := decimal.NewFromFloat(float64(10)).Pow(decimal.NewFromBigInt(new(big.Int).SetBytes(ret.Balances[j].Metadata.Decimals), 0))
-		mkI := priceI.Mul(decimal.NewFromBigInt(new(big.Int).SetBytes(ret.Balances[i].Balance), 0).DivRound(mulI, 18))
-		mkJ := priceJ.Mul(decimal.NewFromBigInt(new(big.Int).SetBytes(ret.Balances[j].Balance), 0).DivRound(mulJ, 18))
-
-		return mkI.Cmp(mkJ) >= 0
+		return bytes.Compare(ret.Balances[i].Token, ret.Balances[j].Token) < 0
```

Comment:
Shouldn't bigtable return the columns anyway in sorted order?

Reply:
For every token we request the metadata within a unique go-routine. Those finish in an undefined order so sorting is needed.